src/libslic3r/GCode.hpp

```diff
@@ -347,8 +347,12 @@ class GCode {
     std::string     preamble();
     // BBS
     std::string     change_layer(coordf_t print_z);
-    std::string     extrude_entity(const ExtrusionEntity &entity, std::string description = "", double speed = -1.);
-    std::string     extrude_loop(ExtrusionLoop loop, std::string description, double speed = -1.);
+    // Orca: pass the complete collection of region perimeters to the extrude loop to check whether the wipe before external loop
+    // should be executed
+    std::string     extrude_entity(const ExtrusionEntity &entity, std::string description = "", double speed = -1., ExtrusionEntitiesPtr region_perimeters = ExtrusionEntitiesPtr());
```

Comment:
It would be better to pass ExtrusionEntitiesPtr by const reference instead to avoid creatign a copy of the vector

Reply:
Done